lib/iron_bank/local_records.rb

```diff
@@ -6,52 +6,91 @@ module IronBank
   class LocalRecords
     private_class_method :new
 
-    RESOURCES = %w[
-      Product
-      ProductRatePlan
-      ProductRatePlanCharge
-      ProductRatePlanChargeTier
-    ].freeze
+    RESOURCE_QUERY_FIELDS = {
+      "Product"                   => ["*"],
+      "ProductRatePlan"           => ["*", "Product.Id"],
+      "ProductRatePlanCharge"     => ["*", "ProductRatePlan.Id"],
+      "ProductRatePlanChargeTier" => ["*", "ProductRatePlanCharge.Id"]
+    }.freeze
 
     def self.directory
       IronBank.configuration.export_directory
     end
 
     def self.export
       FileUtils.mkdir_p(directory) unless Dir.exist?(directory)
-      RESOURCES.each { |resource| new(resource).export }
+      RESOURCE_QUERY_FIELDS.keys.each { |resource| new(resource).save_file }
     end
 
-    def export
-      CSV.open(file_path, "w") do |csv|
-        # first row = CSV headers
-        write_headers(csv)
-        write_records(csv)
+    def save_file
+      until completed? || max_query?
+        IronBank.logger.info(export_query_info)
+        sleep backoff_time
       end
+
+      File.open(file_path, "w") { |file| file.write(export.content) }
     end
 
     private
 
-    attr_reader :resource
+    BACKOFF = {
+      max:                 3,
+      interval:            0.5,
+      interval_randomness: 0.5,
+      backoff_factor:      4
+    }.freeze
+    private_constant :BACKOFF
+
+    attr_reader :resource, :query_attempts
 
     def initialize(resource)
-      @resource = resource
+      @resource       = resource
+      @query_attempts = 0
     end
 
-    def klass
-      IronBank::Resources.const_get(resource)
+    def export
+      @export ||= IronBank::Export.create(
+        "select #{RESOURCE_QUERY_FIELDS[resource].join(', ')} from #{resource}"
+      )
     end
 
     def file_path
       File.expand_path("#{resource}.csv", self.class.directory)
     end
 
-    def write_headers(csv)
-      csv << klass.fields
+    def export_query_info
+      "Waiting for export #{export.id} to complete " \
+        "(attempt #{query_attempts} of #{BACKOFF[:max]})"
+    end
+
+    def completed?
+      return false unless (status = export.reload.status)
+
+      case status
+      when "Pending", "Processing" then false
+      when "Completed"             then true
+      else
+        raise IronBank::Error, "Export #{export.id} has status #{export.status}"
+      end
+    end
+
+    def max_query?
+      @query_attempts += 1
+      return false unless @query_attempts > BACKOFF[:max]
+
+      raise IronBank::Error, "Export query attempts exceeded"
     end
 
-    def write_records(csv)
-      klass.find_each { |record| csv << record.to_csv_row }
+    def backoff_time
```

Comment:
Maybe renaming some locals might make this more clear?
```suggestion
    def backoff_time
      interval          = BACKOFF[:interval]
      factor            = BACKOFF[:backoff_factor]
      randomness        = BACKOFF[:interval_randomness].to_f 
      constant_interval = interval * (factor ** query_attempts)
      random_interval   = rand * randomness * interval

      constant_interval + random_interval
    end
```

Reply:
I think with the removed redundancy in the `BACKOFF` constant, it reads clearly now:

```rb
def backoff_time
  interval         = BACKOFF[:interval]
  current_interval = interval * (BACKOFF[:factor]**query_attempts)
  random_interval  = rand * BACKOFF[:randomness].to_f * interval

  current_interval + random_interval
end
```